File: firstCodebase/clear_database.py
"""
clear_database.py
Utility script to clear all entries in the Firebase database (Python version)
"""

import requests
from firebase_config import COLLECTIONS, FIREBASE_DB_URL

def clear_collection(collection):
    """Clear all entries in a specific collection"""
    try:
        print(f"Clearing all entries in {collection}...")
        url = f"{FIREBASE_DB_URL}/{collection}.json"
        response = requests.delete(url)
        response.raise_for_status()  # Raise an exception for HTTP errors
        print(f"✅ Successfully cleared all entries in {collection}")
    except Exception as error:
        print(f"❌ Error clearing entries in {collection}: {error}")

def clear_all_collections():
    """Clear all entries in all collections"""
    print("🔄 Starting database cleanup...")
    
    # Clear each collection
    clear_collection(COLLECTIONS.SHOOTER_IMAGE)
    clear_collection(COLLECTIONS.SHOOTER_VERIFICATION)
    clear_collection(COLLECTIONS.SHOOTER_COORDINATES)
    
    print("✅ Database cleanup complete! All collections have been cleared.")
    print("You can now start testing with a fresh database.")

if __name__ == "__main__":
    # Run the cleanup
    try:
        clear_all_collections()
    except Exception as error:
        print(f"❌ Error during database cleanup: {error}")

File: firstCodebase/old-code/clear-database.js
/**
 * clear-database.js
 * Utility script to clear all entries in the Firebase database
 */

const { COLLECTIONS } = require('./firebaseConfig');
const { initializeApp } = require('firebase/app');
const { getDatabase, ref, remove } = require('firebase/database');

// Firebase configuration
const firebaseConfig = {
  apiKey: "AIzaSyAtvS-jJZNzn_5T8_y-zkKQazNfONo33SU",
  authDomain: "protechxion-app.firebaseapp.com",
  databaseURL: "https://protechxion-app-default-rtdb.firebaseio.com",
  projectId: "protechxion-app",
  storageBucket: "protechxion-app.firestorage.app",
  messagingSenderId: "1025772388829",
  appId: "1:1025772388829:web:eab2947e46299980d83634",
  measurementId: "G-161S6JPJ38"
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const database = getDatabase(app);

/**
 * Clear all entries in a specific collection
 */
const clearCollection = async (collection) => {
  try {
    console.log(`Clearing all entries in ${collection}...`);
    const collectionRef = ref(database, collection);
    await remove(collectionRef);
    console.log(`✅ Successfully cleared all entries in ${collection}`);
  } catch (error) {
    console.error(`❌ Error clearing entries in ${collection}:`, error);
  }
};

/**
 * Clear all entries in all collections
 */
const clearAllCollections = async () => {
  console.log('🔄 Starting database cleanup...');
  
  // Clear each collection
  await clearCollection(COLLECTIONS.SHOOTER_IMAGE);
  await clearCollection(COLLECTIONS.SHOOTER_VERIFICATION);
  await clearCollection(COLLECTIONS.SHOOTER_COORDINATES);
  
  console.log('✅ Database cleanup complete! All collections have been cleared.');
  console.log('You can now start testing with a fresh database.');
};

// Run the cleanup
clearAllCollections().catch(error => {
  console.error('❌ Error during database cleanup:', error);
});
